### PR TITLE
Ember >= 4.x compatibility (require ember-source 4.8+)

### PR DESCRIPTION
**Pull request by void-mAlex**
motivation for this is that @ember/legacy-built-in-components is a peer dep for ember-engines for "ember-source": "^3.24.1 || 4",
in ember-source: 4 sendAction was [rightly cleaned up](https://github.com/emberjs/ember.js/commit/52a1384a0c8a4cff7cc87d018e19d31188b7507d) and the named export is no longer resolvable which can cause warnings at best and errors at worst (trying to run an ember-engine while using a LinkTo component on ember source 4.x)

with this PR I propose we close that gap by releasing a new major for @ember/legacy-built-in-components to be used with ember-source >= 4.x

### CODE REVIEW

**Comment by SergeAstapov** [comment]
tbh not sure it worth the churn and effort. maybe I'm missing if there is an actual issue this PR fixes?

**Comment by void-mAlex** [comment]
the short answer is I believe it to be worth the effort.
the actual issue is with running engines with embroider under vite. this is an error in that case and your app doesn't load at all because the import it's doing is not valid on ember-source > 3.x
the alternative would be to pr a change to engines to work without LinkTo component for example or any of the components provided by this addon

**Comment by SergeAstapov** [comment]
> the actual issue is with running engines with embroider under vite

sounds reasonable!

> the alternative would be to pr a change to engines to work without LinkTo component for example or any of the components provided by this addon

it means copy-pasting pasts of this addon.
we tried to brainstorm ideas how to move forward without using something old LinkTo component and came up with https://github.com/emberjs/rfcs/pull/819 which was put on hold as there is larger router initiative as part of Polaris.

**Review by SergeAstapov — APPROVED** [review]

**Comment by SergeAstapov** [comment]
I wonder if @mixonic is ok to land this and get new major out.

@void-mAlex I would say, we need to provide peerDeps on `ember-source: >= 4` then + update readme as well as ember-try scenarios?

**Comment by void-mAlex** [comment]
ok will try and get those done

regarding the failing ember-try scenario for ember-beta
I think either adding @glint/template to dev dependencies of the addon or configuring the scenario to install it for that run would fix it 
any preference on the approach?

**Comment by SergeAstapov** [comment]
@void-mAlex either should be fine IMO. Thank you!

**Comment by void-mAlex** [comment]
@mixonic @SergeAstapov I managed to get ci green as well
bumped peer declaration to be ember-source >=4 
updated readme to reflect that
ember-beta is running against 5.x series so the loose peer declaration should be correct/ok to leave open for future ember version

this would unblock engines continuing to work through the 4.x and 5.x ember source version especially through embrodier

let me know if I missed something or anything else I need to do to land this. 
thank you in advance

File: config/ember-try.js
Comment (SergeAstapov) [inline config/ember-try.js]: not needed as listed in `package.json#devDependencies`?

File: package.json
Comment (SergeAstapov) [inline package.json]: should we make this
```suggestion
    "ember-source": ">= 4.8"
```
as we say 4.8+ in readme and only test those versions?

**Review by SergeAstapov — APPROVED** [review]

File: package.json
Comment (void-mAlex) [reply package.json]: applied

File: config/ember-try.js
Comment (void-mAlex) [reply config/ember-try.js]: removed

**Pull request merged**
